modules/imgproc/src/smooth.cpp

```diff
@@ -2819,24 +3282,33 @@ template <typename ET, typename FT>
 void vlineSmooth5N14641(const FT* const * src, const FT*, int, ET* dst, int len)
 {
     for (int i = 0; i < len; i++)
-        dst[i] = (FT::WT(src[2][i])*6 + ((FT::WT(src[1][i]) + FT::WT(src[3][i]))<<2) + FT::WT(src[0][i]) + FT::WT(src[4][i])) >> 4;
+        dst[i] = (FT::WT(src[2][i])*(uint8_t)6 + ((FT::WT(src[1][i]) + FT::WT(src[3][i]))<<2) + FT::WT(src[0][i]) + FT::WT(src[4][i])) >> 4;
 }
 template <>
 void vlineSmooth5N14641<uint8_t, ufixedpoint16>(const ufixedpoint16* const * src, const ufixedpoint16*, int, uint8_t* dst, int len)
 {
     int i = 0;
     v_uint32x4 v_6 = v_setall_u32(6);
-    for (; i < len - 7; i += 8)
+    for (; i < len - 15; i += 16)
```

Comment:
Current OpenCV pattern for such loops is:
```
for (; i <= len - <vec_width>; i += <vec_width>)
{
    ...
}
```

Using of different constants should be avoided (16/15)